### ISSUE TITLE
Support for ContentSearch within documents 

### ISSUE THREAD

**Issue by shraddhabhata** [dadoonet/fscrawler#263]
I have documents stored in S3 - I need to use ElasticSearch for indexing and also for searching within the contents of the files stored in S3. I need a NodeJS example as well. Please help!!

**Comment by dadoonet** [comment]
@lbroudoux Would you be interested in porting your code from https://github.com/lbroudoux/es-amazon-s3-river to here?

If not, I can may be reuse part of the code you wrote at https://github.com/lbroudoux/es-amazon-s3-river/blob/master/src/main/java/com/github/lbroudoux/elasticsearch/river/s3/connector/S3Connector.java

Let me know :)

**Comment by lbroudoux** [comment]
I'll definitely be happy to contribute and port the amazon-s3-river here ! I'll have a look at fscrawler code in forthcoming days to see how to go. 

Do you have any hints on how to bootstrap this contrib ? Any interface to implement, guideline to follow, etc ?

**Comment by dadoonet** [comment]
I'm going to do some refactoring before so it will be easier to add other sources than FS, like S3, Dropbox... 

I'll let you know when ready. 

But thanks for confirming you'd like to work on that :)

**Comment by shraddhabhata** [comment]
Please let me know if there was any update on this?

**Comment by dadoonet** [comment]
Don't expect anything before some weeks or months

**Comment by shraddhabhata** [comment]
checking in .. Please let me know

**Comment by dadoonet** [comment]
This depends on #367

**Comment by mqmurtuza** [comment]
Any updates on S3 code in fscrawler? I am in the process of adding S3 code, but was wondering if you have made any progress on this?

**Comment by dadoonet** [comment]
@mqmurtuza Nope. It needs more preparation like #268 before something like this can happen.
I'd also want to add a Plugin system to make all that more flexible.

This is going to require a huge work and I'm almost sure it won't happen before some months.
In the meantime you can always fork the project and replace the current Local FS implementation or you can try to use the REST interface FSCrawler exposes.

Like:

* Fetch all data from S3 (with s3cmd or whatever)
* For each file, upload it to FSCrawler Rest endpoint.

**Comment by shadiakiki1986** [comment]
[kahing/goofys](https://github.com/kahing/goofys) could be helpful

> Goofys allows you to mount an S3 bucket as a filey system.

**Comment by mqmurtuza** [comment]
I already cracked it, will need to change little bit to make it available
public.

Thanks
Mohammed

On Feb 2, 2018 8:33 AM, "David Pilato" <notifications@github.com> wrote:

@mqmurtuza <https://github.com/mqmurtuza> Nope. It needs more preparation
like #268 <https://github.com/dadoonet/fscrawler/issues/268> before
something like this can happen.
I'd also want to add a Plugin system to make all that more flexible.

This is going to require a huge work and I'm almost sure it won't happen
before some months.
In the meantime you can always fork the project and replace the current
Local FS implementation or you can try to use the REST interface FSCrawler
exposes.

Like:

   - Fetch all data from S3 (with s3cmd or whatever)
   - For each file, upload it to FSCrawler Rest endpoint.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/dadoonet/fscrawler/issues/263#issuecomment-362601187>,
or mute the thread

**Comment by PMarci** [comment]
Could I please have some hints about when this feature can be expected to be merged? I understand there is a large construction site involved in the issue, because of the need for increased abstraction and modularity. As this is an inopportune time to ask for something like this separately, because of the nearing major release, I'd also like to ask whether the suggestion of @mqmurtuza about using the FSCrawler REST endpoint is still the best way to bridge this gap?

**Comment by dadoonet** [comment]
@PMarci Nothing in the short/near term sadly. 

I suggest to try this way: https://github.com/dadoonet/fscrawler/issues/263#issuecomment-362601187
